### PR TITLE
B4b neon

### PR DESCRIPTION
**Pull request by wwieder**
### Description of changes
Fixes neon bugs related to calendar issue with transient cases created by run_neon and an odd error that can be fixed by removing redundant information in usermod_dirs.

### Specific notes
**Confirmed that Contributors other than yourself, if any:**
discussions with @ekluzek and @TeaganKing 

**CTSM Issues Fixed:**
#2470, #2471, #2474

**Are answers expected to change (and if so in what way)?**
No, removes bug in NEON cases

**Any User Interface Changes (namelist or namelist defaults changes)?**
only changes to usermod_dirs

**Testing performed, if any:**
*Manual test*. 
- Using run_neon I confirmed this would create a new transient case (must explicitly set --run-type) 
`./run_neon --neon-site KONA --run-type transient --output-root /glade/derecho/scratch/wwieder/neon_ag`
- I also created a case from scratch that built and ran with the following:
`./create_newcase --case /glade/derecho/scratch/wwieder/neon_ag/HARV.ad --res CLM_USRDAT --compset I1PtClm51Bgc --output-root /glade/derecho/scratch/wwieder/neon_ag --run-unsupported --user-mods-dir NEON/HARV`
- postad case now have run directory for a softlink to restart files, tested with: `./run_neon --neon-sites DEJU --run-type postad --experiment dev175 --output-root /glade/derecho/scratch/wwieder/neon_AK --base-case /glade/derecho/scratch/wwieder/neon_AK/DEJU`

Outstanding issues (that I'm aware of, mentioned in #2433) include:
- [x] The `--experiment` flag no longer seems to work in creating the cases, like the DEJU.postad example above, which should be DEJU.dev175.postad, this is addressed in #2435

### CODE REVIEW

**Comment by wwieder** [comment]
Outstanding issues (that I'm aware of) are:
- explicit need to set the --experiment flag in https://github.com/ESCOMP/CTSM/issues/2433 and
- issue with the run_dir not existing, https://github.com/ESCOMP/CTSM/issues/2437#issuecomment-2019084983

**Comment by wwieder** [comment]
@samsrabin I think you're up for b4b-dev this cycle so I'm assigning this PR to you.

**Comment by wwieder** [comment]
@TeaganKing and @slevis-lmwg did you have a solution for the "run_dir not existing" issue mentioned in #2437?

**Comment by slevis-lmwg** [comment]
@wwieder to see the code change that fixed it for me, start in this directory:
`/glade/work/slevis/git/cn-matrix_v3`
and type the following:
`git diff 0767f1d...25f0236 cime_config/SystemTests/sspmatrixcn.py`
You should get:
```
diff --git a/cime_config/SystemTests/sspmatrixcn.py b/cime_config/SystemTests/sspmatrixcn.py
index bd41d8465..eec462e72 100644
--- a/cime_config/SystemTests/sspmatrixcn.py
+++ b/cime_config/SystemTests/sspmatrixcn.py
@@ -207,6 +207,8 @@ class SSPMATRIXCN(SystemTestsCommon):
                      linkfile = os.path.join(rundir, os.path.basename(item))
                      if os.path.exists(linkfile):
                          os.remove( linkfile )
+                     if not os.path.isdir(rundir):
+                         os.makedirs(rundir)
                      os.symlink(item, linkfile )
```

**Comment by ekluzek** [comment]
@wwieder based on what @TeaganKing found I think you should try what I suggest here...

https://github.com/ESCOMP/CTSM/issues/2437#issuecomment-2053858195

I think that might fix the rundir issue and maybe help with the CALENDER stuff, although it sounds like this PR is a workaround by explicitly setting the CALENDAR for the cases, and some other time/date changes you made...

**Review by ekluzek — APPROVED** [review]
Changes are all fine. It's not obvious that they would fix anything. But, explicitly setting the CALENDAR doesn't hurt anything. And since the changes in the user_nl_clm file aren't needed they should be removed. So that's a good change. It's not obvious why it would matter as they should be valid. But, anytime you have more complexity than is needed -- it's more likely that something will go wrong. So always good to simplify if possible.

**Comment by slevis-lmwg** [comment]
From discussing with @wwieder 
We need this PR merged to main for things to work again.
So I will checkout this branch and run all python testing.
If tests are OK (as expected), then I will merge the branch to the b4b-dev branch.
I suggested to @wwieder to open issue(s) for loose ends that need to be tied (if any).

**Comment by wwieder** [comment]
I realized that @TeaganKing has a fix for the outstanding --experiment issue I mentioned above.  @TeaganKing can you open a PR from your neon_experiment_fix branch to the ctsm/b4b-dev branch.  I think this will resolve issues created with externals update in dev172?  

If this is OK, @slevis-lmwg could you also integrate @TeaganKing's PR onto b4b-dev?

**Comment by slevis-lmwg** [comment]
Testing on derecho:
OK: ./run_ctsm_py_tests -u
IN PROGRESS: ./run_ctsm_py_tests -s
IN PROGRESS: ./run_sys_tests -s clm_pymods -c ctsm5.1.dev175 --skip-generate

**Comment by TeaganKing** [comment]
Hi @wwieder were you referring to [this PR](https://github.com/ESCOMP/CTSM/pull/2435)? It addresses the `experiment` issue due to how the arguments were set in parent/child classes.

**Comment by wwieder** [comment]
Yes, I'd forgotten you already opened it. Thanks @TeaganKing !

**Comment by slevis-lmwg** [comment]
@ekluzek @samsrabin @adrifoster 
what is our protocol for closing resolved issues when merging to b4b-dev?
Can I close these now or should I wait for b4b-dev to come to main?

https://github.com/ESCOMP/CTSM/issues/2470
https://github.com/ESCOMP/CTSM/issues/2471
https://github.com/ESCOMP/CTSM/issues/2474

**Comment by samsrabin** [comment]
We wait for it to come to main. I'll go ahead and add closing keywords for those issues to this PR, so it should happen automatically upon merge to main.

**Comment by samsrabin** [comment]
Thanks for pointing out that gap in the [doc](https://docs.google.com/document/d/17quJfi5trKxoNtY-QOkxwfG-iLcKRROy8HOl80SIpJ0/edit), by the way. I've now added text to that effect.

File: cime_config/usermods_dirs/NEON/defaults/user_nl_clm [hunk 1]
```diff
@@ -18,9 +18,6 @@
 ! Set glc_do_dynglacier  with GLC_TWO_WAY_COUPLING               env variable
 !----------------------------------------------------------------------------------
 
-flanduse_timeseries = ' '   ! This isn't needed for a non transient case, but will be once we start using transient compsets
-fsurdat = "$DIN_LOC_ROOT/lnd/clm2/surfdata_map/NEON/surfdata_1x1_NEON_${NEONSITE}_hist_78pfts_CMIP6_simyr2000_c230601.nc"
```
Comment (slevis-lmwg) [inline cime_config/usermods_dirs/NEON/defaults/user_nl_clm:22]: I have now merged #2472 and #2435; however, believe it or not the very last test on izumi (that was pending and I decided to not wait for) shows a problem that may be due to the removal of fsurdat here...
```
ERROR : CLM build-namelist::CLMBuildNamelist::add_default() : No default value found for fsurdat.
            Are defaults provided for this resolution and land mask?'
```
@wwieder how would you like to address this?

Comment (slevis-lmwg) [reply cime_config/usermods_dirs/NEON/defaults/user_nl_clm:22]: I should have mentioned the test:
`SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.izumi_nag.clm-NEON-MOAB--clm-PRISM.C.0416-112218iz_nag`

Comment (wwieder) [reply cime_config/usermods_dirs/NEON/defaults/user_nl_clm:22]: I'd suggest not removing the surface dataset if that gets the test to pass.  Is that simple enough to add back in?

Comment (slevis-lmwg) [reply cime_config/usermods_dirs/NEON/defaults/user_nl_clm:22]: I will try it tomorrow. If that fixes it, I think simplest is for me to open a new PR with the one-line change and merge into b4b-dev.

**Pull request merged**